fe/fe-core/src/main/java/com/starrocks/qe/VariableMgr.java

```diff
@@ -341,6 +346,83 @@ public static void read(DataInputStream in) throws IOException, DdlException {
         }
     }
 
+    public static void save(DataOutputStream dos) throws IOException, SRMetaBlockException {
+        Map<String, String> m = new HashMap<>();
+        Map<String, String> g = new HashMap<>();
+        try {
+            for (Field field : SessionVariable.class.getDeclaredFields()) {
+                field.setAccessible(true);
+                VarAttr attr = field.getAnnotation(VarAttr.class);
+                if (attr == null) {
+                    continue;
+                }
+                switch (field.getType().getSimpleName()) {
+                    case "boolean":
+                    case "int":
+                    case "long":
+                    case "float":
+                    case "double":
+                    case "String":
+                        m.put(attr.name(), field.get(DEFAULT_SESSION_VARIABLE).toString());
```

Comment:
do we need to filter the `SESSION_ONLY` variables here?

Reply:
done